examples/nidcpower_source_dc_voltage/measurement.py

```diff
@@ -78,89 +83,97 @@ def measure(
+                    # https://github.com/ni/nimi-python/issues/1885
+                    measured_values = []
+                    with channels.initiate():
+                        deadline = time.time() + time_remaining
+                        while True:
+                            if time.time() > deadline:
+                                measurement_service.context.abort(
+                                    grpc.StatusCode.DEADLINE_EXCEEDED, "deadline exceeded"
+                                )
+                            if pending_cancellation:
+                                measurement_service.context.abort(
+                                    grpc.StatusCode.CANCELLED, "client requested cancellation"
+                                )
+                            try:
+                                channels.wait_for_event(
+                                    nidcpower.enums.Event.SOURCE_COMPLETE, timeout=0.1
+                                )
+                                break
+                            except nidcpower.errors.DriverError as e:
+                                """
+                                There is no native way to support cancellation when taking a DCPower
+                                measurement. To support cancellation, we will be calling WaitForEvent
+                                until it succeeds or we have gone past the specified timeout. WaitForEvent
+                                will throw an exception if it times out, which is why we are catching
+                                and doing nothing.
+                                """
+                                if (
+                                    e.code == NIDCPOWER_WAIT_FOR_EVENT_TIMEOUT_ERROR_CODE
+                                    or e.code == NIDCPOWER_TIMEOUT_EXCEEDED_ERROR_CODE
+                                ):
+                                    pass
+                                else:
+                                    raise
+
+                        measured_values = channels.measure_multiple()
+                        for index, mapping in enumerate(channel_mappings):
+                            measured_values[index] = measured_values[index]._replace(
+                                in_compliance=session.channels[
+                                    mapping.channel
+                                ].query_in_compliance()
+                            )
+                    session = None  # Don't abort after this point
+                _log_measured_values(channel_mappings, measured_values)
+                measurement_sites += [m.site for m in channel_mappings]
+                measurement_pin_names += [m.pin_or_relay_name for m in channel_mappings]
+                voltage_measurements += [m.voltage for m in measured_values]
+                current_measurements += [m.current for m in measured_values]
+                in_compliance += [m.in_compliance for m in measured_values]
```

Comment:
I thought one of the reasons to write this helper function was so that clients wouldn't need to deal with channel mappings directly.

Reply:
There is some details of channel mapping being added into the session info object we are returning from our new API.- https://github.com/ni/measurementlink-python/blob/f974731efa6ea622802aa060ab481461356c0a5d/ni_measurementlink_service/session_management.py#L152
I think we could include the required details here and not return the entire channel mappings structure.

Reply:
This has been resolved and i understand the comment now